optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/rotation/TimeBucket.java

```diff
@@ -0,0 +1,210 @@
+        this.endTime = endTime;
+        this.additionalSkillSet = additionalSkillSet;
+        this.repeatOnDaySet = repeatOnDaySet;
+        this.seatList = new ArrayList<>(rotationLength);
+        DayOfWeek rotationDay = startOfWeek;
+        for (int i = 0; i < rotationLength; i++) {
+            if (repeatOnDaySet.contains(rotationDay)) {
+                seatList.add(new Seat(i,null));
+            }
+            rotationDay = rotationDay.plus(1);
+        }
+    }
+    
+    public TimeBucket(Integer tenantId, Spot spot, LocalTime startTime, LocalTime endTime,
+                      Set<Skill> additionalSkillSet,
+                      Set<DayOfWeek> repeatOnDaySet, List<Seat> seatList) {
+        super(tenantId);
+        this.spot = spot;
+        this.startTime = startTime;
+        this.endTime = endTime;
+        this.additionalSkillSet = additionalSkillSet;
+        this.repeatOnDaySet = repeatOnDaySet;
+        this.seatList = new ArrayList<>(seatList);
+    }
+    
+    
+    public Spot getSpot() {
+        return spot;
+    }
+    
+    
+    public void setSpot(Spot spot) {
+        this.spot = spot;
+    }
+
+    public LocalTime getStartTime() {
+        return startTime;
+    }
+
+    
+    public void setStartTime(LocalTime startTime) {
+        this.startTime = startTime;
+    }
+
+    
+    public LocalTime getEndTime() {
+        return endTime;
+    }
+
+    
+    public void setEndTime(LocalTime endTime) {
+        this.endTime = endTime;
+    }
+
+    
+    public Set<Skill> getAdditionalSkillSet() {
+        return additionalSkillSet;
+    }
+
+    
+    public void setAdditionalSkillSet(Set<Skill> additionalSkillSet) {
+        this.additionalSkillSet = additionalSkillSet;
+    }
+
+    
+    public Set<DayOfWeek> getRepeatOnDaySet() {
+        return repeatOnDaySet;
+    }
+
+    
+    public void setRepeatOnDaySet(Set<DayOfWeek> repeatOnDaySet) {
+        this.repeatOnDaySet = repeatOnDaySet;
+    }
+
+    
+    public List<Seat> getSeatList() {
+        return seatList;
+    }
+
+    
+    public void setSeatList(List<Seat> seatList) {
+        this.seatList = seatList;
+    }
+
+    public void setValuesFromTimeBucket(TimeBucket updatedTimeBucket) {
+        setSpot(updatedTimeBucket.getSpot());
+        setStartTime(updatedTimeBucket.getStartTime());
+        setEndTime(updatedTimeBucket.getEndTime());
+        setAdditionalSkillSet(updatedTimeBucket.getAdditionalSkillSet());
+        setRepeatOnDaySet(updatedTimeBucket.getRepeatOnDaySet());
+        setSeatList(updatedTimeBucket.getSeatList());   
+    }
+    
+    public Optional<Shift> createShiftForOffset(LocalDate startDate, int offset, ZoneId zoneId,
+                                                boolean defaultToRotationEmployee) {
+        Optional<Seat> maybeSeat = seatList.stream().filter(seat -> seat.getDayInRotation() == offset).findAny();
+        
+        if (!maybeSeat.isPresent()) {
+            return Optional.empty();
+        }
```

Comment:
Although I like the use of `Optional`, the `isPresent()` makes things a bit ugly.
I suggest refactoring this method so that `.findAny().map()` can be used. Don't be afraid of extracting some of the logic into another method.

Reply:
Neat; didn't know Optional have map